images/argocd-repo-server/tests/main.tf

```diff
@@ -0,0 +1,46 @@
+terraform {
+  required_providers {
+    oci  = { source = "chainguard-dev/oci" }
+    helm = { source = "hashicorp/helm" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_exec_test" "run" {
+  digest = var.digest
+  script = "docker run --rm $${IMAGE_NAME} --help"
+}
+
+resource "random_string" "random" {
+  length  = 6
+  upper   = false
+  special = false
+}
+
+resource "helm_release" "argocd-repo-server" {
+  depends_on = [data.oci_exec_test.run]
+
+  name = "argocd-repo-server-${random_string.random.result}"
+
+  repository = "https://argoproj.github.io/argo-helm"
+  chart      = "argo"
+
+  namespace        = "argocd-repo-server-${random_string.random.result}"
+  create_namespace = true
+  skip_crds        = true
```

Comment:
nit: leave a note here so this argocd specific incantation doesn't get copy/pasta'd everywhere